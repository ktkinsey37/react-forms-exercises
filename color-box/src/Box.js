import React from "react";



function Box(props) {

  // console.log(props, props.box.color, "THIS IS THE LINE THAT IS BOTH LOGGING AND THROWING?")

  return (
    <div>
          <div
        className="box"
        key={props.id}
        style={{
          backgroundColor: `${props.box.color}`,
          position: "relative",
          height: `${props.box.height}px`,
          width: `${props.box.width}px`,
          border: "2px solid black"
        }}
      >
      </div>
      <button onClick={props.remove}>X</button>
    </div>

  );
}

export default Box;
