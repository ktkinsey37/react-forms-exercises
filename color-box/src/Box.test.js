import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

const box = {color:"green", width:"50", height:"50", id:"0"}


it("matches snapshot", function() {
  const { asFragment } = render(<Box box={box} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing", function() {
  render(<Box box={box} />);
});