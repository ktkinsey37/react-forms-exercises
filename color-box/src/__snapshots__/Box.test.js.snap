// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="box"
      style="background-color: green; position: relative; height: 50px; width: 50px; border: 2px solid black;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
